libs/common/include/launchdarkly/data/evaluation_detail.hpp

```diff
@@ -52,21 +60,33 @@ class EvaluationDetail {
      */
     [[nodiscard]] std::optional<EvaluationReason> const& Reason() const;
 
+    /**
+     * @return True if the evaluation resulted in an error.
+     */
+    [[nodiscard]] bool IsError() const;
+
     /**
      * @return A reference to the variation value.
      */
     T const& operator*() const;
 
+    /**
+     * @return True if the evaluation was successful (i.e. IsError returns
+     * false.)
+     */
+    explicit operator bool() const;
+
```

Comment:
Unclear to me if we need both `IsError` and `operator bool()`. 

The intention is to allow users (and us, in test code) to check if the evaluation has an error, without having to reach deep into the object. 

Having `if (detail)` is pretty convenient and fits in with `std::optional` and `tl::expected`, I just don't know if we need the explicit `IsError` as well.

Reply:
std::optional, and tl::expected both have kind of pointer like behavior, and they both additionally have an explicit way to check for a value `has_value`. We don't really have pointer like behavior, so I don't know that the boolean operator would be very apparent. It isn't bad to have, but an error and having a value are a bit different for an evaluation reason.

Reply:
It's true it's not pointer-like, but we do have `operator*` to get at the value. So you could do:
```c++
 if (result) {
      double x = (*result).AsDouble()
 }
```
But this does feel clunky. Hmm.

Reply:
Perhaps the `VariationDetail` methods should really accept detail as an out-param.

Reply:
Generally speaking I don't think you usually even care if there was an error.